dal/src/main/java/com/epam/esm/dal/CertificateDao.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.esm.dal;
+
+import java.util.List;
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.transferobj.FilterParam;
+import com.epam.esm.transferobj.OrderParam;
+
+public interface CertificateDao {
+
+	GiftCertificate addCertificate(GiftCertificate certificate);
```

Comment:
In my view, since this is CertificateDao, it is obvious that main operations work with certificates and there is no necessity to add entity name to every method name, moreover, you have appropriate name of the parameter.